packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -513,20 +483,66 @@ export async function downloadSnapshot(
     fetchBinarySnapshotFormat?: boolean,
     controller?: AbortController,
     epochTracker?: EpochTracker,
+    disableTreesLatestForBinaryWireFormat?: boolean,
 ): Promise<ISnapshotRequestAndResponseOptions> {
     // back-compat: This block to be removed with #8784 when we only consume/consider odsp resolvers that are >= 0.51
     const sharingLinkToRedeem = (odspResolvedUrl as any).sharingLinkToRedeem;
     if(sharingLinkToRedeem) {
         odspResolvedUrl.shareLinkInfo = { ...odspResolvedUrl.shareLinkInfo, sharingLinkToRedeem };
     }
 
-    if (fetchBinarySnapshotFormat) {
-        // Logging an event here as it is not supposed to be used in production yet and only in experimental mode.
-        logger.sendTelemetryEvent({ eventName: "BinarySnapshotFetched" });
+    // If fetchBinarySnapshotFormat is true but we don't want to use trees latest for this, then we go into this flow
+    // of fetching binary wire format with /content api.
+    if (fetchBinarySnapshotFormat && disableTreesLatestForBinaryWireFormat) {
```

Comment:
So, it sounds like we have two controls - one by service (whatever format it returns, when client supports both) and one by client (this code here). Is this mostly for flexibility to make decisions?

I'd suggest not to have two booleans (as it's very confusing how their combination works, without looking at code). I'd suggest having an enum with these values:
- Use V1
- Use V2
- Use V1 or V2, depending on what service says.

Reply:
So, V2 can be fetched either by contents/attachments api(temporary api until we permanently move to trees latest for fetching binary snapshot) or by trees/latest using the accept header.
With this change we are going to use trees latest by default for fetching binary snapshot however I added a feature gate so that if there is some problem with accept header, we can move back to using the attachment api for fetching binary snapshot and we can keep testing in testing phase.
If this does not sound good, then we can remove "disableTreesLatestForBinaryWireFormat" and also remove contents/attachments api from the code in this file with this change and just always use trees/latest for both the versions.

Reply:
Got it. I'd still suggest using enum vs. 2 booleans, as you only have 3 states, not 4 (2x2 independent states).
It's almost always better that way :)

Reply:
After some more thinking, I think I am going to delete the older attachment api as the snapshot parsing code is still the same. It was just about the accept header and after testing it seems good and we shouldn't need to have a separate feature gate for this.